test/test_foreach.py

```diff
@@ -85,26 +85,39 @@ def _test_unary_op(self, device, dtype, foreach_op, foreach_op_, torch_op):
                 self.assertEqual(tensors1, expected)
 
     def _test_pointwise_op(self, device, dtype, foreach_op, foreach_op_, torch_op):
+        value = 2
         for N in N_values:
             tensors = self._get_test_data(device, dtype, N)
             tensors1 = self._get_test_data(device, dtype, N)
             tensors2 = self._get_test_data(device, dtype, N)
-            value = 2
 
-            # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
-            control_dtype = torch.float32 if (self.device_type == 'cuda' and
-                                              (dtype is torch.float16 or dtype is torch.bfloat16)) else dtype
-            expected = [torch_op(tensors[i].to(dtype=control_dtype),
-                                 tensors1[i].to(dtype=control_dtype),
-                                 tensors2[i].to(dtype=control_dtype), value=value).to(dtype=dtype) for i in range(N)]
+            for values in [value, [value for _ in range(N)]]:
```

Comment:
We should test something where not all of the Scalar list values are the same to check that the kernels are passing the Scalar list around correctly